OmiseSDK/AtomeFormViewController.swift

```diff
@@ -0,0 +1,333 @@
+    @IBOutlet private var shippingPostalCodeErrorLabel: UILabel!
+    
+    @IBOutlet var formLabels: [UILabel]!
+    @IBOutlet var formFields: [OmiseTextField]!
+    
+    @IBOutlet var formFieldsAccessoryView: UIToolbar!
+    @IBOutlet var gotoPreviousFieldBarButtonItem: UIBarButtonItem!
+    @IBOutlet var gotoNextFieldBarButtonItem: UIBarButtonItem!
+    @IBOutlet var doneEditingBarButtonItem: UIBarButtonItem!
+    
+    // need to refactor loadView, removing super results in crash
+    // swiftlint:disable prohibited_super_call
+    override func loadView() {
+        super.loadView()
+        
+        view.backgroundColor = .background
+        formFieldsAccessoryView.barTintColor = .formAccessoryBarTintColor
+        
+        submitButton.defaultBackgroundColor = .omise
+        submitButton.disabledBackgroundColor = .line
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        applyPrimaryColor()
+        applySecondaryColor()
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+        
+        formFields.forEach {
+            $0.inputAccessoryView = formFieldsAccessoryView
+        }
+        
+        formFields.forEach {
+            $0.adjustsFontForContentSizeCategory = true
+        }
+        formLabels.forEach {
+            $0.adjustsFontForContentSizeCategory = true
+        }
+        submitButton.titleLabel?.adjustsFontForContentSizeCategory = true
+        
+        if  #available(iOS 11, *) {
+            // We'll leave the adjusting scroll view insets job for iOS 11 and later to the layoutMargins + safeAreaInsets here
+        } else {
+            automaticallyAdjustsScrollViewInsets = true
+        }
+        
+        phoneNumberTextField.addTarget(self, action: #selector(validateForm), for: .editingDidEndOnExit)
+        emailTextField.addTarget(self, action: #selector(validateForm), for: .editingDidEnd)
+        shippingStreetTextField.addTarget(self, action: #selector(validateForm), for: .editingDidEnd)
+        shippingCountryCodeTextField.addTarget(self, action: #selector(validateForm), for: .editingDidEnd)
+        shippingCityTextField.addTarget(self, action: #selector(validateForm), for: .editingDidEnd)
+        shippingPostalCodeTextField.addTarget(self, action: #selector(validateForm), for: .editingDidEnd)
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillChangeFrame(_:)),
+            name: NotificationKeyboardWillChangeFrameNotification,
+            object: nil
+        )
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillHide(_:)),
+            name: NotificationKeyboardWillHideFrameNotification,
+            object: nil
+        )
+
+        phoneNumberTextField.validator = try? NSRegularExpression(pattern: "^[0-9+]{0,1}+\\d{10,11}?", options: [])
```

Comment:
How about -> `(?:(?:\\+\\d{2})|0)?\\d{9}` (a plus sign following with 2 digits, or zero then 9 digits after). 
I'm not sure this will work 😂 but you can try this out krub

Reply:
Changed as https://github.com/omise/omise-ios/pull/184/commits/833358db2d693cc567123cf9020a1b033544e3db ka. (also tested as you mention `a plus sign following with 2 digits, or zero then 9 digits after.`

Reply:
Cool! 👍